model/jpa/src/main/java/org/keycloak/models/jpa/session/PersistentUserSessionEntity.java

```diff
@@ -32,16 +32,32 @@
  * @author <a href="mailto:mposolda@redhat.com">Marek Posolda</a>
  */
 @NamedQueries({
-        @NamedQuery(name="deleteUserSessionsByRealm", query="delete from PersistentUserSessionEntity sess where sess.realmId = :realmId"),
+        @NamedQuery(name="deleteUserSessionsByRealm", query="delete from PersistentUserSessionEntity sess where sess.realmId = :realmId and sess.offline = coalesce(:offline, sess.offline) "),
```

Comment:
This leads currently to seq scan of the table.

Reply:
The query `deleteUserSessionsByRealm` results in the following SQL:
```sql
delete from OFFLINE_USER_SESSION sess
where sess.realm_id = 'acme'
  and sess.offline_flag = coalesce(cast(null as text), sess.offline_flag)
```
Without modifications this results in a full table scan as you mentioned.
![image](https://user-images.githubusercontent.com/314690/107160595-835fbd80-6997-11eb-871a-d865b56134c3.png)

With the following index:
```
<createIndex tableName="OFFLINE_USER_SESSION" indexName="IDX_OFFLINE_USS_CLEANUP">
            <column name="OFFLINE_FLAG" type="VARCHAR(4)"/>
            <column name="realm_id" type="VARCHAR(36)"/>
        </createIndex>
```
the SQL statement uses the index and is much faster:
![image](https://user-images.githubusercontent.com/314690/107160624-a7230380-6997-11eb-9774-25b7be54cede.png)